src/message_conversions.rs

```diff
@@ -0,0 +1,530 @@
+            if ue_request_correlation.get(&app_request_id).is_none() {
+                ue_request_correlation.insert(app_request_id, req_id.clone());
+                trace!("{} - (app_request_id, req_id)  inserted for later correlation in UE_REQUEST_CORRELATION: ({}, {})",
+                UP_CLIENT_VSOMEIP_FN_TAG_CONVERT_UMSG_TO_VSOMEIP_MSG,
+                    app_request_id, req_id.to_hyphenated_string(),
+                );
+            } else {
+                return Err(UStatus::fail_with_code(UCode::ALREADY_EXISTS, format!("Already exists same request with id: {app_request_id}, therefore rejecting")));
+            }
+            get_pinned_message_base(&vsomeip_msg).set_return_code(vsomeip::return_code_e::E_OK);
+            let payload = {
+                if let Some(bytes) = umsg.payload.clone() {
+                    trace!("payload is set, it's: {bytes:?}");
+                    bytes.to_vec()
+                } else {
+                    trace!("payload is not set");
+                    Vec::new()
+                }
+            };
+            trace!("therefore, payload is: {payload:?}");
+            let mut vsomeip_payload =
+                make_payload_wrapper(get_pinned_runtime(runtime_wrapper).create_payload());
+            set_data_safe(get_pinned_payload(&vsomeip_payload), &payload);
+            set_message_payload(&mut vsomeip_msg, &mut vsomeip_payload);
+
+            let request_id = get_pinned_message_base(&vsomeip_msg).get_request();
+            let service_id = get_pinned_message_base(&vsomeip_msg).get_service();
+            let client_id = get_pinned_message_base(&vsomeip_msg).get_client();
+            let session_id = get_pinned_message_base(&vsomeip_msg).get_session();
+            let method_id = get_pinned_message_base(&vsomeip_msg).get_method();
+            let instance_id = get_pinned_message_base(&vsomeip_msg).get_instance();
+            let interface_version = get_pinned_message_base(&vsomeip_msg).get_interface_version();
+
+            trace!("{} - : request_id: {} client_id: {} session_id: {} service_id: {} instance_id: {} method_id: {} interface_version: {} app_client_id: {}",
+                UP_CLIENT_VSOMEIP_FN_TAG_CONVERT_UMSG_TO_VSOMEIP_MSG,
+                request_id, client_id, session_id, service_id, instance_id, method_id, interface_version, app_client_id
+            );
+
+            Ok(vsomeip_msg)
+        }
+        UMessageType::UMESSAGE_TYPE_RESPONSE => {
+            trace!(
+                "{} - Attempting to send Response",
+                UP_CLIENT_VSOMEIP_FN_TAG_CONVERT_UMSG_TO_VSOMEIP_MSG,
+            );
+
+            let mut vsomeip_msg =
+                make_message_wrapper(get_pinned_runtime(runtime_wrapper).create_message(true));
+
+            let (_instance_id, service_id) = split_u32_to_u16(source.ue_id);
+            get_pinned_message_base(&vsomeip_msg).set_service(service_id);
+            let instance_id = 1; // TODO: Setting to 1 manually for now
```

Comment:
IIRC from reading the SOME/IP spec, I _do_ need to respect the upper four bytes of the ue_id and assign the instance_id appropriately according to them, right? Tagging @stevenhartley & @int0x27